sonar-plugin-api/src/main/java/org/sonar/api/batch/rule/CheckFactory.java

```diff
@@ -26,7 +26,9 @@
  * by dependency injection. It must not be extended by plugins.
  *
  * @since 4.2
+ * @deprecated since 6.0 sonar-check-api is deprecated
  */
+@Deprecated
 @BatchSide
 public class CheckFactory {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)